api/src/main/java/jakarta/data/repository/Sort.java

```diff
@@ -57,28 +57,35 @@
  * <li>the database is incapable of ordering with the requested
  *     sort criteria.</li>
  * </ul>
+ *
+ * @param property    name of the property to order by.
+ * @param isAscending whether ordering for this property is ascending (true) or descending (false).
+ * @param ignoreCase  whether or not to request case insensitive ordering from a database with case sensitive collation.
  */
-public final class Sort {
-
-    private final String property;
-
-    private final Direction direction;
+public record Sort(String property, boolean isAscending, boolean ignoreCase) {
 
-    private final boolean ignoreCase;
-
-    private Sort(String property, Direction direction, boolean ignoreCase) {
-        this.property = property;
-        this.direction = direction;
-        this.ignoreCase = ignoreCase;
+    /**
+     * Defines sort criteria for an entity property.
+     *
+     * @param property    name of the property to order by.
+     * @param isAscending whether ordering for this property is ascending (true) or descending (false).
+     * @param ignoreCase  whether or not to request case insensitive ordering from a database with case sensitive collation.
+     */
+    public Sort {
+        Objects.requireNonNull(property, "property is required");
     }
 
+    // Override to provide method documentation:
     /**
+     * Name of the property to order by.
+     *
      * @return The property name to order by; will never be {@literal null}.
      */
     public String property() {
-        return this.property;
+        return property;
```

Comment:
Do we need this method?
Once you don't overwrite, you can remove it.

Reply:
We will get the same implementation from record if we remove it, but the documentation will be lost.  I have a comment just above the method clarifying why it is there, `Override to provide method documentation:`